llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -175,21 +177,32 @@ MDNode *attributeToDecorateMetadata(LLVMContext &Ctx, const Attribute &Attr) {
 /// Tries to generate a SPIR-V execution mode metadata node from an attribute.
 /// If the attribute is unknown \c None will be returned.
 ///
-/// @param M     [in] the LLVM module.
+/// @param F     [in] the LLVM function.
 /// @param Attr  [in] the LLVM attribute to generate metadata for.
 ///
 /// @returns a pair with the name of the resulting metadata and a pointer to
 ///          the metadata node with its values if the attribute has a
 ///          corresponding SPIR-V execution mode. Otherwise \c None is returned.
 std::optional<std::pair<std::string, MDNode *>>
-attributeToExecModeMetadata(Module &M, const Attribute &Attr) {
+attributeToExecModeMetadata(Function &F, const Attribute &Attr) {
```

Comment:
Not related to this PR, but just general comment about the code: it seems a bit weird to me that the main input of this function is `Attr` to translate to, but it accepts a Module/Function first. 

It looks especially weird with a function, because why do we need an attribute then if we already have a function? Perhaps it is better to swap the order, to simply indicate that function is some kind of extra context/payload here. I definitely don't suggest to do this in this PR, because it is too unrelated, this is just a though about the code.

Reply:
do we have any process for tracking work like this so it doesn't get lost? in the past i've seen them tracked as 'techncial debt' issues. if so im happy to make one for this.